flytekit/bin/entrypoint.py

```diff
@@ -104,7 +105,9 @@ def _dispatch_execute(
         idl_input_literals = _literal_models.LiteralMap.from_flyte_idl(input_proto)
 
         # Step2
-        outputs = task_def.dispatch_execute(ctx, idl_input_literals)
+        # Decorate the dispatch execute function before calling it, this wraps all exceptions into one
+        # of the FlyteScopedExceptions
+        outputs = _scoped_exceptions.system_entry_point(task_def.dispatch_execute)(ctx, idl_input_literals)
```

Comment:
omg, this is weird? 
why not a simple
try:
   outputs = ....
except :
  ...

Reply:
This is effectively what the decorator does.  The reason the decorator exists is mostly for two reasons:
1. We enter and leave user-code in multiple places, 
2. The excpetions that we, flytekit authors, throw do not conform to any hierarchy easily distinguishable from potential user exceptions.  After we complete https://github.com/flyteorg/flyte/issues/1033, and all flytekit exceptions have a base class that users will never use, we can get rid of these scoped exceptions and just use a few try catches to distinguish things.

I think it's preferable to get rid of these scoped exceptions actually, I just don't want to go through everything and update them right now.  Maybe at 1.0